Update ore dictionary

Zenith-One opened the pull request:
I've added the following entries to the ore dictionary:

- Prismarine Shard ( `"gemPrismarine"` )
- Prismarine Crystals ( `"dustPrismarine"` )
- Prismarine Block ( `"blockPrismarine"` )
- Prismarine Brick ( `"blockPrismarineBrick"` )
- Dark Prismarine ( `"blockDarkPrismarine"` )
- Nether Wart ( `"cropNetherWart"` )
- Sugarcane ( `"sugarcane"` )
- Paper ( `"paper"` )
- Ender Pearl ( `"enderpearl"` )
- Bone ( `"bone `)
- Gunpowder ( `"gunpowder `)
- String ( `"string"` )
- Nether Star ( `"netherStar"` )
- Leather ( `"leather"` )
- Feather ( `"feather"` )
- Egg ( `"egg"` )
- End Stone ( `"endstone"` )
- Torch ( `"torch"` )

I've also organized the registration and replacement code.

As suggested below, I brought in #2306 and #2480 as well in order to consolidate additions to default OD entries, which, combined, add the following entries:

- Vine ( `"vine"` )
- Cactus ( `"blockCactus"` )
- Grass ( `"grass"`)
- Obsidian ( `"obsidian"` )
- Red Sandstone ( added to `"sandstone"` )
- Crafting Table ( `"workbench"` )
- Slime Block ( `"blockSlime"` )
- Granite ( `"stoneGranite"` )
- Polished Granite ( `"stoneGranitePolished"` )
- Diorite ( `"stoneDiorite"` )
- Polished Diorite ( `"stoneDioritePolished"` )
- Andesite ( `"stoneAndesite"` )
- Polished Andesite ( `"stoneAndesitePolished"` )

kashike commented on src/main/java/net/minecraftforge/oredict/OreDictionary.java:
```diff
@@ -77,43 +81,57 @@ public static void initVanillaEntries()
-            registerOre("slimeball",   Items.slime_ball);
-            registerOre("glowstone",   Blocks.glowstone);
+
+            // storage blocks
```
**storage** blocks?

Zenith-One replied:
That's what I've always heard iron/gold/diamond/whatever blocks called. Each block represents 9 ingots worth, which makes for convenient storage.

If there's a more generally accepted name for that kind of block I don't mind updating

elifoster replied:
Indeed. That's what we call it on the wiki, if that makes you feel any better.

williewillus commented:
And to elaborate, these are added to allow forge's OD recipe replacements to propagate to more items by default

Zenith-One commented:
Not all of the ore dict entries I added have associated recipes, but they seemed like things which mods might want to add more kinds of and use in recipes. For instance prismarine blocks aren't used in any minecraft recipes, but I can imagine them being used in mod recipes, and I can imagine modders adding new variants of prismarine blocks as well.

Adaptivity commented:
@Zenith-One you should probably merge changes from PRs #2306 and #2480 into this one.

williewillus commented:
yeah go ahead and roll them all here

LexManos commented:
Do you have a list of all the recipes that are modified by the new additions.
You should be able to fairly easily print them out in the replacement code.
We need to be sure that the replacements are OVER reaching in recipes and making them not match vanilla.

immibis commented:
Why do we not have a more generic item replacement dictionary, where any item can be made to substitute for any other in recipes?

That way instead of having to eventually add every vanilla item to the ore dictionary (what if someone wants to ore-dict wheat seeds, or books?), we could use their item names. And it would affect mod recipes that don't explicitly use the ore dictionary.

Zenith-One commented:
Merged in the other PRs referenced above.

LexManos commented:
Lets see im concerned about a few entries:

>tile.stone.dioriteSmooth.name
>tile.stone.graniteSmooth.name
>tile.stone.andesiteSmooth.name
>tile.stone.granite.name

What changes did you make that added those in, the wiki is telling me the recipes are not effected by any of those items....

blockDarkPrismarine -> blockPrismarineDark To match the others.

LexManos commented on src/main/java/net/minecraftforge/oredict/OreDictionary.java:
```diff
@@ -126,33 +146,107 @@ public static void initVanillaEntries()
+            registerOre("torch",       Blocks.torch);
+            registerOre("blockSlime",    Blocks.slime_block);
+            registerOre("blockPrismarine",
+                    new ItemStack(Blocks.prismarine, 1, BlockPrismarine.EnumType.ROUGH.getMetadata()));
```
No need for wrap.

LexManos commented on src/main/java/net/minecraftforge/oredict/OreDictionary.java:
```diff
@@ -126,33 +146,107 @@ public static void initVanillaEntries()
             registerOre("chestWood",   Blocks.chest);
             registerOre("chestEnder",  Blocks.ender_chest);
             registerOre("chestTrapped", Blocks.trapped_chest);
+            registerOre("workbench",   Blocks.crafting_table);
```
Crafting table, is not a chest... Its also not in your inital post about what you're changing.

Zenith-One commented:
The diorite, andesite, and granite entries are made from #2480 which would explain those recipes being affected. Workbench, and anything else that I didn't specify in the initial PR comment are from either #2306 or #2480 

I pulled those into this PR as suggested by @Adaptivity and @williewillus 

I've moved the workbench up with the rest of the blocks, changed `"blockDarkPrismarine"` to '"blockPrismarineDark"` and removed the superfluous line breaks in the prismarine registration. the lines.

LexManos commented:
Update your inital message to include all the things you change.

Adaptivity commented:
i guess this is pulled into the wrong branch. Anyway it is missing in 1.9 branch

LexManos commented:
SONOVA When I fist saw it it was on the 1.9 branch. How did he screw that up?
@Zenith-One Fix your PR.

matthewprenger commented:
GitHub doesn't let you re-target PR branches, so this one has to have always been targeting master.

LexManos commented:
Well then i derped up, :/ We need a 1.9 PR.

matthewprenger commented:
@LexManos Looks like you can cherry-pick it with no conflicts.
```
git checkout 1.9
git cherry-pick d31cf8ddbc52c9d5d9792762fd7d06a756848ff3
git push
```
